modules/RTC/TraceablePeerConnection.js

```diff
@@ -2240,15 +2240,25 @@ TraceablePeerConnection.prototype._adjustRemoteMediaDirection = function(remoteD
     const transformer = new SdpTransformWrap(remoteDescription.sdp);
 
     [ MediaType.AUDIO, MediaType.VIDEO ].forEach(mediaType => {
-        const media = transformer.selectMedia(mediaType)?.[0];
-        const hasLocalSource = this.hasAnyTracksOfType(mediaType);
-        const hasRemoteSource = this.getRemoteTracks(null, mediaType).length > 0;
-
-        media.direction = hasLocalSource && hasRemoteSource
-            ? MediaDirection.SENDRECV
-            : hasLocalSource
-                ? MediaDirection.RECVONLY
-                : hasRemoteSource ? MediaDirection.SENDONLY : MediaDirection.INACTIVE;
+        const media = transformer.selectMedia(mediaType);
+        const localSources = this.getLocalTracks(mediaType).length;
+        const remoteSources = this.getRemoteTracks(null, mediaType).length;
+
+        media.forEach((mLine, idx) => {
+            if (localSources && localSources === remoteSources) {
```

Comment:
I am lost what we are doing here with this change. Could you please explain? Maybe adding a comment will be good?

Reply:
Before, we were setting the direction only on the first audio and video m-line. Now, we are setting on each of the m-lines based on how many local and remote sources are present.